pkg/noderesourcetopology/score_test.go

```diff
@@ -536,6 +472,144 @@ func TestNodeResourceScorePluginLeastNUMA(t *testing.T) {
 	}
 }
 
+// when only a subset of nodes has NRT data available[1], prefer the nodes which have the NRT data over the other nodes;
+// IOW, a node without NRT data available should always have score == 0
+func TestNodeResourcePartialDataScorePlugin(t *testing.T) {
+	type podRequests struct {
+		pod        *v1.Pod
+		name       string
+		wantStatus *framework.Status
+	}
+	pRequests := []podRequests{
+		{
+			pod: makePodByResourceList(&v1.ResourceList{
+				v1.ResourceCPU:    *resource.NewQuantity(2, resource.DecimalSI),
+				v1.ResourceMemory: *resource.NewQuantity(20*1024*1024, resource.DecimalSI)}),
+			name:       "Pod1",
+			wantStatus: nil,
+		},
+	}
+
+	type testScenario struct {
+		name      string
+		wantedRes nodeToScoreMap
+		requests  []podRequests
+		strategy  scoreStrategyFn
+		nrtFilter nrtFilterFn
+	}
+
+	tests := []testScenario{
+		{
+			name:      "No data at all, MostAllocated strategy",
+			wantedRes: nodeToScoreMap{},
+			requests:  pRequests,
+			strategy:  mostAllocatedScoreStrategy,
+			nrtFilter: func(nrt *topologyv1alpha2.NodeResourceTopology) *topologyv1alpha2.NodeResourceTopology {
+				return nil
+			},
+		},
+		{
+			name:      "No data at all, LeastAllocated strategy",
+			wantedRes: nodeToScoreMap{},
+			requests:  pRequests,
+			strategy:  leastAllocatedScoreStrategy,
+			nrtFilter: func(nrt *topologyv1alpha2.NodeResourceTopology) *topologyv1alpha2.NodeResourceTopology {
+				return nil
+			},
+		},
+		{
+			name:      "No data at all, BalancedAllocation strategy",
+			wantedRes: nodeToScoreMap{},
+			requests:  pRequests,
+			strategy:  balancedAllocationScoreStrategy,
+			nrtFilter: func(nrt *topologyv1alpha2.NodeResourceTopology) *topologyv1alpha2.NodeResourceTopology {
+				return nil
+			},
+		},
+		{
+			name:      "One node with NRT data, MostAllocated strategy",
+			wantedRes: nodeToScoreMap{"Node1": 27},
```

Comment:
it took me a while to spot you are leveraging the Go famous default value of 0 for nonexistent keys in map 😄